### PR TITLE
Add new Energy Summary Table Component

### PR DESCRIPTION
**Pull request by ldodds**
Extracts the existing summary table used on the adult dashboard, the print view of the adult dashboard and the total energy use into a standalone component. Added some specs around it which slightly improves the coverage we have already for checking the table is displaying expected data.

This is integrated into the current dashboards and pages.

To support the styling for the dashboard I've altered the component so that it adds classes where needed. Have slightly tidied up some of the existing CSS for that table too.

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -84,6 +84,7 @@
   config.include ViewComponent::SystemTestHelpers, type: :component
   config.include Capybara::RSpecMatchers, type: :component
   config.include Devise::Test::ControllerHelpers, type: :component
+  config.include Devise::Test::ControllerHelpers, type: :helper
```
Comment (tbhi) [inline spec/rails_helper.rb:87]: https://github.com/heartcombo/devise#controller-tests looks to say Devise::Test::IntegrationHelpers should be used for newer rails

File: spec/helpers/help_page_helper_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-describe HelpPageHelper do
+describe HelpPageHelper, type: :helper do
```
Comment (tbhi) [inline spec/helpers/help_page_helper_spec.rb:3]: my rubocop thinks this isn't neede as inferred - https://docs.rubocop.org/rubocop-rspec_rails/cops_rspecrails.html#rspecrailsinferredspectype

File: spec/components/energy_summary_table_component_spec.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  end
+
+  context 'when school is data enabled' do
+    it 'renders the table title' do
+      expect(html).to have_content(I18n.t('schools.show.summary_of_recent_energy_usage'))
+    end
+
+    it 'renders the table' do
+      expect(html).to have_content(I18n.t('classes.tables.summary_table_data.last_year'))
+      expect(html).to have_content('1,000')
```
Comment (tbhi) [inline spec/components/energy_summary_table_component_spec.rb:53]: could be good to use `have_selector(:table_row, {})`

**Review by tbhi — APPROVED** [review]

File: spec/rails_helper.rb (hunk 1 above)
Comment (ldodds) [reply spec/rails_helper.rb:87]: That doesn't work though. The error without this, or with `IntegrationHelpers` has:

```
If you are seeing this on one of your tests, ensure that your tests are either executing the Rails middleware stack or that your tests are using the `Devise::Test::ControllerHelpers` module to inject the `request.env['warden']` object for you.
```

Comment (tbhi) [reply spec/rails_helper.rb:87]: my mistake - rspec looks to be Devise::Test::ControllerHelpers only

**Pull request merged**